src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -353,19 +397,29 @@ await CallWithRequestTracing(
                     {
                         watchedSettings = new Dictionary<KeyValueIdentifier, ConfigurationSetting>(_watchedSettings);
 
-                        foreach (KeyValueWatcher changeWatcher in refreshableWatchers.Concat(refreshableMultiKeyWatchers))
+                        foreach (KeyValueWatcher changeWatcher in refreshableWatchers.Concat(refreshableFeatureFlagWatchers))
                         {
                             UpdateNextRefreshTime(changeWatcher);
                         }
 
-                        foreach (KeyValueChange change in keyValueChanges.Concat(changedKeyValuesCollection))
+                        if (_options.RefreshOptions.RegisterAllEnabled && registerAllIsRefreshable)
+                        {
+                            _registerAllNextRefreshTime = DateTimeOffset.UtcNow.Add(_options.RefreshOptions.RefreshInterval);
```

Comment:
I think we should remove `FeatureFlagWatchers ` and do the same thing for feature flags since they are all based on collection monitoring.